### DISCUSSION TITLE
Amazon Warehousing and Distribution

### DISCUSSION THREAD

**Discussion by habberr** [amzn/selling-partner-api-models#3430] [General] [answered]
Hi!

There is this service that Amazon provides — **Amazon Warehousing and Distribution** (AWD). As far as I understand, you can ship your products to AWD warehouse, where it can be stored for some time until you FBA stock for this product runs low, at which point you can order to ship this product from AWD to FBA. 

I haven't been able to find any mentions of it in the SP API documentation. It seems that FulfillmentInbound and FBAInventory API endpoints do not provide information on stock on AWD and inbound transfers to AWD. 

Do you know if it possible to obtain AWD transfers/stock information via SP API or any other way?

**Comment by ritikacd** [comment, marked as answer]
Hi habberr,
AWD Transfers or getting stock information is not available via SP-API yet, but API product team has been working on developing and testing APIs involved in transfers and maintaining inventory. They are expected to release around Q2 2023 for General availability.

**Reply by habberr** [reply]
Hi ritikacd,
Thank you for the info.
Do I understand correclty that as of now AWD data is not available through Reports either?

**Reply by ritikacd** [reply]
That's correct! There's no report for the AWD APIs yet, may I know how will you use reports if available for AWD data?

**Reply by habberr** [reply]
We're operating several brands, so we want to integrate this data into our ERP system as it affects replenishment process.

**Reply by ritikacd** [reply]
Thank you for providing the insights, we will pass this data to AWD API product team. Also, we will keep you posted on the AWD APIs availability once launched.

**Reply by DanielLanger** [reply]
any update on release timeline?

**Comment by grantrosse** [comment]
Hey what is the timeline for this? We are also looking to pull this data.

**Comment by aaalekseev** [comment]
Happy New Year and up to the topic - 2024 came..

**Comment by ahmedfeki** [comment]
Q1 2024 almost over any updates ? 
We need this now as many fees can be waived if we use AWD

**Comment by ShenAdrada** [comment]
any updates on this?

**Reply by sdepablos** [reply]
Any updates on this?

**Comment by HeGuanyuan** [comment]
Today, I saw that the AWD API has been released: https://developer-docs.amazon.com/sp-api/docs/awd_2024-05-09-reference. However, it mentions:

> Prerequisites
> To complete this tutorial, you need:
> 1. Authorization from the selling partner for whom you are making calls. For more information, refer to Authorizing Selling Partner API Applications.
> 2. The AWD role assigned to your developer profile.
> 3. The AWD role selected in the app registration page for your application.


The AWD Role mentioned in points 2 and 3 cannot be found in the developer and app configuration. It is possible that the seller central has not been updated yet.

**Reply by chengpan9** [reply]
I also encountered the same problem, did you find a solution?

**Reply by ritikacd** [reply]
Hi @HeGuanyuan and @chengpan9 

The AWD role is expected to go live tomorrow(6/21) for Selling Partners and Developers. You should be able to assign AWD role to your developer profile and, once approved, select it in the app registration page for your application to use the AWD APIs.

Thanks, 
Ritika
Selling Partner API Team

**Reply by ritikacd** [reply]
Hi @HeGuanyuan and @chengpan9
The AWD role is live! You should be see it on your developer profile page and be able to select it. 

Thanks,
Ritika
Selling Partner API Team

**Reply by chengpan9** [reply]
Thanks for your reply, I can now choose the AWD role, thanks again!

**Reply by mschwartz19** [reply]
seems that the API only has options for viewing, is there a way to create shipments from API?or is there an expected release date for that?

**Comment by J-W-Chan** [comment]
What is the difference between this api and the [inbound api](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#getshipment)?

**Reply by cohlar** [reply]
This API refers to AWD shipments whereas the [inbound api](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#getshipment) refers to FBA shipments.

**Comment by mschwartz19** [comment]
seems that the API only has options for viewing, is there a way to create shipments from API?
or is there an expected release date for that?

**Reply by ritikacd** [reply]
Hello @mschwartz19 

Thank you for proposing an enhancement to the AWD APIs capabilities. We are currently working on implementing this change, however, we do not have an estimated time of when the implementation will be complete. As soon as this occurs, we will post an announcement in the [Announcements Section](https://developer-docs.amazon.com/sp-api/changelog), please subscribe to our [RSS feeds](https://developer-docs.amazon.com/sp-api/changelog.rss) to get notified for updates. 

Thanks,
Ritika Chaudhary
Selling Partner API Developer Services

**Comment by omrishaked** [comment]
@ritikacd I still can't find the role permissions.  What am I doing wrong?
<img width="702" alt="image" src="https://github.com/user-attachments/assets/f6b55168-e3c2-4778-a7ca-2ae6a3811c0d">

**Reply by ritikacd** [reply]
@omrishaked Please ensure [Amazon Warehousing and Distribution role](https://developer-docs.amazon.com/sp-api/docs/amazon-warehousing-distribution-role) is assigned to your developer profile. Apply the role in the developer profile and submit the form. Once you have approved access to the role, it will show up in the app registration page for your application. Then, you should be able to select it. Make sure you create a new refresh token to access the APIs associated with the new role.

**Reply by tgehrs** [reply]
For anyone else wondering where to assign a role to your developer profile, you can do so here https://sellercentral.amazon.com/developer/register, then navigate to the [list of applications](https://sellercentral.amazon.com/sellingpartner/developerconsole) and click "Edit App" to view the page that @omrishaked screenshotted above
